src/autotester/server/utils/tests/test_file_management.py

```diff
@@ -0,0 +1,176 @@
+        return_value=fake_redis,
+    ):
+        yield fake_redis
+
+
+@pytest.fixture(autouse=True)
+def tmp_script_dir():
+    """
+    Mock the test_script_directory method and yield a temporary directory
+    """
+    with tempfile.TemporaryDirectory() as tmp_dir:
+        files_dir = os.path.join(tmp_dir, "files")
+        os.mkdir(files_dir)
+        with tempfile.NamedTemporaryFile(dir=files_dir):
+            with patch(
+                "autotester.server.utils.redis_management.test_script_directory",
+                return_value=tmp_dir,
+            ):
+                yield tmp_dir
+
+
+@pytest.fixture
+def t_path():
+    """
+    Returns a temporary directory with one sub directory
+    """
+    with tempfile.TemporaryDirectory() as tests_path:
+        with tempfile.NamedTemporaryFile(dir=tests_path):
+            yield tests_path
+
+
+@pytest.fixture
+def f_path():
+    """
+    Returns a temporary directory with one sub directory
+    """
+    files_path = tempfile.mkdtemp()
+    f_d, files = tempfile.mkstemp(dir=files_path)
+    return files_path
+
+
+@pytest.fixture
+def args():
+    """
+    Returns markus address and assignment id
+    """
+    markus_address = "http://localhost:3000/csc108/en/main"
+    assignment_id = 1
+    return markus_address, assignment_id
+
+
+def fd_permission(file_or_dir):
+    """
+    Gets file or directory and returns its permission
+    """
+    mode = os.stat(file_or_dir).st_mode
+    permission = stat.filemode(mode)
+    return permission
+
+
+class TestSetupFiles:
+    def test_group_owner(self, t_path, f_path, args):
+        """
+        Checks whether the group owner of both
+        student files and script files changed into test_username
+        """
+        markus_address, assignment_id = args
+        tests_path = t_path
+        files_path = f_path
+        test_username = Path(tests_path).owner()
+        student_files, script_files = setup_files(
+            files_path, tests_path, test_username, markus_address, assignment_id
+        )
+        group_stfile = []
+        group_scfile = []
+        for _fd, file_or_dir in student_files:
+            group_stfile.append(Path(file_or_dir).group())
+        for _fd, file_or_dir in script_files:
+            group_scfile.append(Path(file_or_dir).group())
+        assert all(test_username == i for i in group_scfile)
+        assert all(test_username == i for i in group_stfile)
+
+    def test_student_files(self, t_path, f_path, args):
+        """
+        Checks whether the permission of files and directories
+        in student files changed into '0o770' and '0o660'
+        """
+        markus_address, assignment_id = args
+        tests_path = t_path
+        files_path = f_path
+        test_username = Path(tests_path).owner()
+        student_files, script_files = setup_files(
+            files_path, tests_path, test_username, markus_address, assignment_id
+        )
+        stdir_pmsn = []
```

Comment:
You're actually to to remove this variable based on my above comment, but I should point out that you should not use abbreviations like "pmsn" in variable names.